src/main/java/org/betonquest/betonquest/menu/utils/Utils.java

```diff
@@ -56,4 +64,78 @@ public static Method getMethod(final Class<?> clazz, final String name, final in
         } while ((superClazz = superClazz.getSuperclass()) != null);
         throw new NoSuchMethodException("Can't find method " + name + " with " + paramlength + " parameters");
     }
+
+    /**
+     * This Method resolves global variables in the menu section
+     * No need for other variable resolving here
+     *
+     * @param input the string to parse
+     * @param pack  the context of the global variables
+     * @return resolved String
+     */
+    public static String resolveGlobalVariables(final String input, final QuestPackage pack) {
+        if (input == null) {
+            return null;
+        }
+        String variableInput = input;
+
+        final Pattern globalVariableRegex = Pattern.compile("\\$([^ $\\s]+)\\$");
+        while (true) {
+            final Matcher matcher = globalVariableRegex.matcher(variableInput);
+            if (!matcher.find()) {
+                break;
+            }
+            final String varName = matcher.group(1);
+            final String varVal;
+            if ("this".equals(varName)) {
+                varVal = pack.getQuestPath();
+            } else {
+                final GlobalVariableID variableID;
+                try {
+                    variableID = new GlobalVariableID(pack, varName);
+                } catch (final ObjectNotFoundException e) {
+                    LOG.warn(pack, e.getMessage(), e);
+                    return variableInput;
+                }
+                final String varRaw = variableID.getPackage().getConfig().getString("variables." + variableID.getBaseID());
+                if (varRaw == null) {
+                    LOG.warn(pack, String.format("Variable %s not defined in package %s", variableID.getBaseID(), variableID.getPackage().getQuestPath()));
+                    return variableInput;
+                }
+                varVal = resolve(varRaw, variableID.getPackage());
+            }
+            variableInput = variableInput.replace("$" + varName + "$", varVal);
+        }
```

Comment:
While this code does what it is supposed to do it is very readable.

A possible way to improve on that would be to use the `Matcher.
appendReplacement()` method:
```java
StringBuilder output = new StringBuilder();
while (matcher.find()) {
  String variableValue = loadGlobalVariable(matcher.group(1), pack);
  matcher.appendReplacement(output, variableValue);
}
return matcher.appendTail(output).toString();
```
The `loadGlobalVariable()` method contains all the glrobal variable resolving logic.

To preserve the efficiency that repeating global variables aren't loaded twice one can use an object:
```java
class VariableLoader {
  QuestPackage pack;
  Map<String, String> values;
  // constructor and initialization omitted
  String loadVariable(String name) {
    return values.computeIfAbsent(name, this::resolveVariable);
  }
  String resolveVariable(String name); // implementation skipped
}
```
Just initialize that object at the beginning (outside the loop) and use it instead of the `loadGlobalVariable()` method from my first example.